bson/src/test/unit/org/bson/codecs/pojo/ClassModelTest.java

```diff
@@ -55,91 +62,82 @@ public void testSimpleGenericsModel() {
         assertEquals("_t", classModel.getDiscriminatorKey());
         assertEquals("org.bson.codecs.pojo.entities.SimpleGenericsModel", classModel.getDiscriminator());
         assertNull(classModel.getIdPropertyModel());
-        assertEquals(4, classModel.getPropertyModels().size());
         assertTrue(classModel.getInstanceCreatorFactory() instanceof InstanceCreatorFactoryImpl);
+
+        assertEquals(4, classModel.getPropertyModels().size());
```

Comment:
Lots of changes to improve the testing of all the type data for the property models.